### PR TITLE
Version 5.9, brought to you by the letter à

### PR DESCRIPTION
**Pull request by rspeer**
- Tweaked the heuristic to decode, for example, `"Ã&nbsp;"` as the letter "à" more often.

- This combines with the non-breaking-space fixer to decode `"Ã "` as `"à"` as well. However, in many cases, the text `" Ã "` was intended to be `" à "`, preserving the space -- the underlying mojibake had two spaces after it, but the Web coalesced them into one. We detect this case based on common French and Portuguese words, and preserve the space when it appears intended.

- Updated the data file of Unicode character categories to Unicode 13, as used in Python 3.9.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -33,13 +33,13 @@
     packages=['ftfy', 'ftfy.bad_codecs'],
     install_requires=['wcwidth'],
     tests_require=['pytest'],
-    python_requires='>=3.5',
+    python_requires='>=3.6',
     classifiers=[
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.5",
```
Comment (Tahnan) [inline setup.py:39]: Not really knowing how people use changelogs: is "no longer supports Python 3.5" something worth noting in the changelog?  Or will people get the information they need from `python_requires` and the lack of this line?

**Review by Tahnan — COMMENTED** [review]
Other than the one comment, this looks good to me.  My only other question is about the French/Catalan tradeoff.  It's unquestionably the case that "à" is more common in French than any word containing "à" will be in any other language, so prioritizing that case does make sense, but the effect on Catalan will go beyond just words that start with an à, like "àudio": looking at the top words in `wordfreq.iter_wordlist('ca')` suggests that fairly common words such as [`gràcies`](https://ca.wiktionary.org/wiki/gr%C3%A0cies) (86th) and [`fàcil`](https://ca.wiktionary.org/wiki/f%C3%A0cil) (672nd) will also end up with a space after the à (also "ràdio", "càrrec", "democràcia"...).

Again, there's very likely more French text than Catalan text in the world, and *many* more instances of French "à" than Catalan "gràcies", so I'll leave it up to you to decide whether this issue is sufficently...grave.

**Comment by rspeer** [comment]
I should change the documentation, given that it affects 'à' in the middle of a word.

I would be worried if this were changing all instances of 'fàcil' to 'fà cil', but what reassures me is that this only applies in the case where it was mojibaked, such as 'fÃ cil'. In that case, I think we have license to take our best guess at the word and its spacing, and I find 'fà cil' to still be an improvement over 'fÃ cil'.

**Comment by rspeer** [comment]
The status of Python 3.5 support, I believe, turns out to be the same as the previous version: it'll actually work there, but the tests fail because of one f-string in `build_data.py`, a script that yells at you if you run it on anything but the latest version of Python anyway.

I could acknowledge the sorta-but-not-really support for Python 3.5, or I could change the f-string.

**Comment by Tahnan** [comment]
As far as 3.5 goes: if the f-string is in a script that's only supposed to run on 3.9 anyway, I wouldn't give it another thought.  I guess the question is whether, in that case, you want to revert the removal from setup.py; if it works just as well in 3.5 as it did before, maybe setup.py should still claim the same thing.

Also, fundamentally agree on the "fÃ cil" point; this change is certainly not breaking anything in Catalan that was working before.  If you want to update the documentation, I'll wait to merge until youv'e done so.

**Pull request merged**